islandora.api.php

```diff
@@ -736,12 +746,22 @@ function hook_cmodel_pid_islandora_derivative() {
 /**
  * Allows for the altering of defined derivative functions.
  */
-function hook_islandora_derivative_alter(&$derivatives, AbstractObject $object) {
+function hook_islandora_derivative_alter(&$derivatives, AbstractObject $object, $ds_modified_params = array()) {
   foreach ($derivatives as $key => $derivative) {
     if ($derivative['destination_dsid'] == 'TN') {
       unset($derivatives[$key]);
     }
   }
+  // Example of altering out derivative generation if only a specified set of
+  // datastream parameters have been modified.
+  $mask = array(
+    'label' => NULL,
+    'dateLastModified' => NULL,
+    'dsid' => NULL,
+  );
+  if (empty(array_diff_key($ds_modified_params, $mask))) {
```

Comment:
@qadan this won't work. empty only takes variables, sadly not functions as parameters (even when the function returns something)